fastembed/image/onnx_image_model.py

```diff
@@ -67,47 +71,85 @@ def onnx_embed(self, images: List[ImageInput], **kwargs) -> OnnxOutputContext:
+
+        num_workers = parallel
+
+        if not providers and cuda and device_ids is not None:
+            num_workers = min(parallel, len(device_ids))
+
+        start_method = "forkserver" if "forkserver" in get_all_start_methods() else "spawn"
+        params = {
+            "model_name": model_name,
+            "cache_dir": cache_dir,
+            "providers": providers,
+            **kwargs,
+        }
+
+        pool = ParallelWorkerPool(
+            num_workers,
+            cls._get_worker_class(),
+            cuda=cuda,
+            device_ids=device_ids,
+            start_method=start_method,
+        )
+        for batch in pool.ordered_map(iter_batch(images, batch_size), **params):
+            yield from batch
+
     def _embed_images(
         self,
         model_name: str,
         cache_dir: str,
         images: ImageInput,
         batch_size: int = 256,
         parallel: Optional[int] = None,
+        providers: Optional[Sequence[OnnxProvider]] = None,
+        cuda: bool = False,
+        device_ids: Optional[List[int]] = None,
         **kwargs,
     ) -> Iterable[T]:
         is_small = False
 
-        if (
-            isinstance(images, str)
-            or isinstance(images, Path)
-            or (isinstance(images, Image.Image))
-        ):
+        if isinstance(images, (str, Path, Image.Image)):
             images = [images]
             is_small = True
 
-        if isinstance(images, list):
-            if len(images) < batch_size:
-                is_small = True
-
-        if parallel == 0:
-            parallel = os.cpu_count()
+        if isinstance(images, list) and len(images) < batch_size:
+            is_small = True
 
         if parallel is None or is_small:
             for batch in iter_batch(images, batch_size):
                 yield from self._post_process_onnx_output(self.onnx_embed(batch))
         else:
-            start_method = "forkserver" if "forkserver" in get_all_start_methods() else "spawn"
-            params = {"model_name": model_name, "cache_dir": cache_dir, **kwargs}
-            pool = ParallelWorkerPool(
-                parallel, self._get_worker_class(), start_method=start_method
+            yield from self._embed_images_parallel(
+                model_name,
+                cache_dir,
+                images,
+                batch_size,
+                parallel,
+                providers,
+                cuda,
+                device_ids,
+                **kwargs,
             )
-            for batch in pool.ordered_map(iter_batch(images, batch_size), **params):
-                yield from self._post_process_onnx_output(batch)
 
 
 class ImageEmbeddingWorker(EmbeddingWorker):
     def process(self, items: Iterable[Tuple[int, Any]]) -> Iterable[Tuple[int, Any]]:
         for idx, batch in items:
             embeddings = self.model.onnx_embed(batch)
-            yield idx, embeddings
+            yield idx, self.model._post_process_onnx_output(embeddings)
```

Comment:
feels like post process should be in the place it was before (for batch in pool.ordered_map ... : yield from post_process(batch))